src/ui_utils.c

```diff
@@ -1022,6 +1022,42 @@ void ui_sidebar_show_hide(void)
 }
 
 
+void ui_menubar_show_hide(void)
```

Comment:
I don't really like this function's implementation.  Usually our `show_hide()` calls don't behave like `toggle()`: you can call them as many times as you want and it'll just make sure stuff is consistent.

So I'd rather see a either caller passing the requested value (e.g. result of `gtk_check_menu_item_get_active()`, or `! ui_prefs.menubar_visible`) as parameter (like `ui_widget_show_hide()`) and the function deciding if it's OK and synchronizing what needs to be; or the caller having to manually toggle `ui_prefs.menubar_visible` and use that as teh "parameter" above.

What do you think?

Reply:
Yes, makes sense totally. I'll think about what could be the best solution.

Reply:
@b4n Check the latest commit - is that more or less what you had in mind?

Reply:
Yes, thanks!